packages/testing/src/web/mockRequests.ts

```diff
@@ -67,16 +67,19 @@ export const registerHandler = (handler: RequestHandler) => {
   }
 }
 
-export type DataFunction = (
-  variables: Record<string, any>,
+export type DataFunction<
+  Query extends Record<string, unknown> = Record<string, unknown>,
```

Comment:
I added the `extends Record<string, unknown>` constraint this morning as I realized we still need to constrain this when no type argument is passed so that `Query` isn't inferred as something like `number`, e.g.:

```ts
    // Still does not allow non-Record<string, unknown> data
    // Argument of type 'number' is not assignable to parameter of type
    // 'Record<string, unknown> | DataFunction<Record<string, unknown>, Record<string, any>>'.
    mockGraphQLMutation('ExampleQuery', 42) // ❌
```

This should align with the previous behavior.